messaging/src/main/java/org/axonframework/messaging/deadletter/QueueIdentifier.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2010-2022. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.messaging.deadletter;
+
+/**
+ * Interface describing an identifier for queues maintained by a {@link DeadLetterQueue}.
+ * <p>
+ * Describes a {@link #identifier()} of the queue, as well as a {@link #group()} the queue belongs to. The
+ * {@code identifier} may, for example, refer to a sequence of messages, whereas the {@code group} can describe a
+ * specific message processing component.
+ *
+ * @author Steven van Beelen
+ * @see DeadLetterQueue
+ * @since 4.6.0
+ */
+public interface QueueIdentifier extends Comparable<QueueIdentifier> {
```

Comment:
Should this class be renamed to `SequenceIdentifier`?
Note that the sequence argument mainly holds for events. Not sure whether sequence is feasible for possible command/query/message-x DLQ's.

Reply:
`QueueIdentifier` seems to suggest there are multiple queues, that said, `SequenceIdentifier` is very generic. Maybe `QueueSequenceIdentifier`?

Reply:
`QueueIdentifier` works for me. In the case of events, the `QueueIdentifier` is primarily based on the `sequenceIdentifier` of the events. In case of commands, it would be based on the `routingKey`.
I'd say it's a matter of context :). From the context of the Queue, `QueueIdentifier` makes perfect sense.

Reply:
Renamed it to `SequenceIdentifier`, as that more closely aligns with the main driving force for having this identifier class in the first place, which is to deviate between sequences (which again differ per group).